lib/spanner/config/semantic_validator.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Spanner.Config.SemanticValidator do
+
+  alias Piper.Permissions.Ast
+  alias Piper.Permissions.Parser
+
+  @moduledoc """
+  Validates bundle config semantics.
+  """
+
+
+  @doc """
+  Accepts a config map and validates semantics. Returns `:ok` if the config is
+  valid and `{:error, err}` on error.
+  """
+  def validate(config) do
+    verify_rules(config)
+  end
+
+  defp verify_rules(config) do
+    permission_set = MapSet.new(config["permissions"])
+
+    Enum.with_index(config["rules"])
+    |> Enum.reduce([], fn({rule, index}, acc) ->
+      {:ok, %Ast.Rule{}=expr, rule_perms} = Parser.parse(rule)
+      [bundle, command] = String.split(expr.command, ":", parts: 2)
+
+      if config["bundle"]["name"] != bundle do
+        acc = [{"The bundle name '#{bundle}' does not match the name in the bundle", "#/rules/#{index}"}  | acc]
+      end
+      if not(command) in Enum.map(config["commands"], &Map.fetch!(&1, "name")) do
```

Comment:
I'm not sure how I missed that. I know I tested this. Let me confirm, but yeah I agree.